### PR TITLE
[RDY] Fix price units in gas stations (fix #18835)

### LINKED ISSUE #18835: Gas station prices need fixing for volume changes
Gas prices ("per unit") currently end up being per mL, which is definitely wrong - $100 gets you maybe 0.3L of gas. I'm not sure what the old "unit" was (250mL? 1L?) but it needs to be adjusted back to that.

### PR DESCRIPTION
**Pull request by BrettDong**
Fixes #18835 . The original price is $3.5 per _original_ unit (250mL). Now we need to multiply volume by 250 to get the correct price.

### CODE REVIEW

**Comment by BrettDong** [comment]
Yep. Currently the player could only buy integral multiple of 250mL of gasoline.

**Comment by BrettDong** [comment]
Yep. Currently the player could only buy integral multiple of 250mL of gasoline.

**Comment by illi-kun** [comment]
Did you test your fix? With this PR, you're not able to buy all gas from gas console (max value you able to buy is available units / 250).

File: src/iexamine.cpp [hunk 1]
```diff
@@ -3021,6 +3021,10 @@ static tripoint getGasPumpByNumber(const tripoint &p, int number)
 
 static bool toPumpFuel(const tripoint &src, const tripoint &dst, long units)
 {
+    // Multiplies units by 250
+    // the new unit is 1mL while the old unit is 250mL
+    units *= 250;
```
Comment (codemime) [inline src/iexamine.cpp]: There's the `units::legacy_volume_factor` for this purpose.

**Comment by illi-kun** [comment]
Closing to reduce line of PRs. Feel free to implement a proper fix.

**Comment by BrettDong** [comment]
OK. Could be reopened.

**Comment by BrettDong** [comment]
@illi-kun

**Comment by illi-kun** [comment]
Done and sorry for delay.

**Comment by BrettDong** [comment]
This PR is ready.

**Pull request merged**